### ISSUE TITLE
API: How to tell a vessel is in local control.

### ISSUE THREAD

**Issue by SirDiazo** [RemoteTechnologiesGroup/RemoteTech#337]
Alright, I just spent a bunch of time with the API and I'm stumped on how I tell a vessel is in local control.

It appears that a vessel under local control has no incoming links to the RT Network, causing weirdness as described below.

The two methods I am looking at are GetShortestSignalDelay and HasAnyConnection

First, GetShortestSignalDelay returns the control delay of the current vessel, however as a vessel under local control has no "signal delay" because it has no incoming link, GetShortestSignalDelay returns Infinity for a vessel under local control, the same as it does for a vessel with no connection.

Second, HasAnyConnection also returns false for a vessel under local control. Again I assume this is because the vessel is under local control, so it has no connection assigned so the method is correctly returning false even though the vessel is controllable. 

Because of this, from my external mod in code, I can not tell the difference between a vessel with no connection and one under local control.

My test vessel for this was a stock Kerbal 2 sitting on the launch pad, I didn't even launch it and the signal display for remotetech in the top left corner was displaying N/A as I was in local control.

Did I miss something here or is this an oversight in the current API?

**Comment by erendrake** [comment]
I would guess that it is an oversight ☺

**Comment by Peppie23** [comment]
What about `HasFlightComputer(Guid id)`?

**Comment by SirDiazo** [comment]
That does not work either, HasFlightComputer returns false on my Kerbal2 on the launchpad that has local control and returns true on a my sattelite in orbit under control with a delay.

So I do not believe this will work for my purposes.

**Comment by Peppie23** [comment]
Hi @SirDiazo ,
i think too, thats the best way. But maybe you can combine and make a own "hasLocalControl" on your source with:
```csharp
Vessel.parts.Any(p => p.isControlSource && p.protoModuleCrew.Any())
```
Maybe add a extra condition to the Api.HasFlightComputer as well.

I'll add the FeatureRequest label to make a new api function like "HasLocalControl" to reference to our `HasLocalControl`-Value for the feature.

**Comment by Peppie23** [comment]
@erendrake i reviewed your new method `HasLocalControl` and i'll get false on a local controlled vessel without an antenna, thats the situation where the deltaTime fragment shows N/A. I think we should change your first condition:
```c#
var satellite = RTCore.Instance.Satellites[id];
if (satellite == null) return false;
```
to
```c#
var satellite = RTCore.Instance.Satellites[id];
if (satellite == null) return true;
```

What do you mean?

**Comment by Peppie23** [comment]
btw i added with #350 your method to the debug window.

**Comment by erendrake** [comment]
My bad then, my test craft had an antenna. I guess i should have tested it with more craft. Expect a patch asap.

**Comment by Peppie23** [comment]
Merged into 1.7
